src/sinks/gcp/stackdriver_logging.rs

```diff
@@ -115,87 +139,62 @@ impl SinkConfig for StackdriverConfig {
     }
 }
 
-impl StackdriverConfig {
-    fn service(
-        &self,
-        cx: &SinkContext,
-        creds: &Option<GcpCredentials>,
-    ) -> crate::Result<RouterSink> {
-        let batch = self.batch.unwrap_or(bytesize::kib(5000u64), 1);
-        let request = self.request.unwrap_with(&REQUEST_DEFAULTS);
-        let encoding = self.encoding.clone();
-        let tls_settings = TlsSettings::from_options(&self.tls)?;
-        let creds = creds.clone();
-
-        // We need to cap the maximum length of the encoded request, so
-        // we encode each event into JSON separately and then splice the
-        // result into a relatively short wrapper.
-        let wrapper = self.write_request();
-        let wrapper_splice = wrapper
-            .find(SPLICE_MAGIC)
-            .expect("Unexpected encoded wrapper format")
-            + SPLICE_OFFSET;
-
-        let build_request = move |mut logs: Vec<u8>| {
-            logs.pop(); // Strip the trailing comma
-
-            let mut body = wrapper.clone().into_bytes();
-            body.splice(wrapper_splice..wrapper_splice, logs);
-
-            let mut request = make_request(body);
-            if let Some(creds) = creds.as_ref() {
-                creds.apply(&mut request);
-            }
-
-            request
-        };
+impl HttpSink for StackdriverSink {
+    type Input = serde_json::Value;
+    type Output = Vec<BoxedRawValue>;
 
-        let http_service = HttpBatchService::new(cx.resolver(), tls_settings, build_request);
+    fn encode_event(&self, mut event: Event) -> Option<Self::Input> {
+        self.config.encoding.apply_rules(&mut event);
 
-        let sink = request
-            .batch_sink(HttpRetryLogic, http_service, cx.acker())
-            .batched_with_min(Buffer::new(false), &batch)
-            .with_flat_map(move |event| iter_ok(Some(encode_event(event, &encoding))));
+        let entry = serde_json::json!({
+            "jsonPayload": event.into_log(),
+        });
 
-        Ok(Box::new(sink))
+        Some(entry)
     }
 
-    fn healthcheck(
-        &self,
-        cx: &SinkContext,
-        creds: &Option<GcpCredentials>,
-    ) -> crate::Result<Healthcheck> {
-        let mut request = make_request(Body::from(self.write_request().into_bytes()));
+    fn build_request(&self, events: Self::Output) -> http::Request<Vec<u8>> {
+        let events = serde_json::json!({
+            "log_name": self.config.log_name(),
+            "entries": events,
+            "resource": {
+                "type": self.config.resource.type_,
+                "labels": self.config.resource.labels,
+            }
+        });
```

Comment:
I preferred the serde approach (with defined structs) but this is definitely simpler.

Reply:
Yeah, the big reason I went with this is that the structs are useful if you use them i many places but if you are not then it becomes noisy.